pkg/controller/cloud/node_controller.go

```diff
@@ -363,7 +363,7 @@ func (cnc *CloudNodeController) AddCloudNode(obj interface{}) {
 
 		curNode.Spec.Taints = excludeTaintFromList(curNode.Spec.Taints, *cloudTaint)
 
-		_, err = cnc.kubeClient.CoreV1().Nodes().Update(curNode)
+		_, err = nodeutil.PatchNode(cnc.kubeClient.CoreV1(), types.NodeName(node.Name), node, curNode, "status")
```

Comment:
Just tested this call will update metadata and status, but not `spec.taints`. Would you please double check?
After the call, it appears:
- 'instance-type' label under metadata was added.
- preparePatchBytes will generate `"taints":null` in payload
- 'node.cloudprovider.kubernetes.io/uninitialized' taint still appears in node spec, which is intended to be removed.

Reply:
Maybe we should specify taints and providerID but I'm not sure, I'll take a look tomorrow.

Reply:
previously, this was updating the metadata+spec of the node (`Nodes().Update(curNode)`)

by patching the "status" subresource, this changes to only update metadata+status, which will not allow updating anything in spec. looking at the code above, it's clear this is attempting to modify taints, so remove the `"status"` subresource from the PatchNode call

Reply:
@liggitt - this what I've done in the first place, but from @jhorwit2's [comment](https://github.com/kubernetes/kubernetes/pull/56350#issuecomment-353176335) seems that GCE adds a status condition.

Reply:
submitting to the status endpoint will not let you update spec.taints, which is what this code is trying to do

Reply:
@liggitt - Yes to me we should only patch the spec.taints. @jhorwit2 what is status condition that GCE is trying to set ?